api/coreservice_with_height.go

```diff
@@ -0,0 +1,71 @@
+package api
+
+import (
+	"context"
+
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/iotexproject/iotex-core/v2/action"
+	"github.com/iotexproject/iotex-core/v2/blockchain/genesis"
+	"github.com/iotexproject/iotex-core/v2/pkg/log"
+	"github.com/iotexproject/iotex-core/v2/pkg/tracer"
+	"github.com/iotexproject/iotex-core/v2/pkg/util/byteutil"
+)
+
+type (
+	// CoreServiceReaderWithHeight is an interface for state reader at certain height
+	CoreServiceReaderWithHeight interface {
```

Comment:
could you explain why we need to create such an interface?

Reply:
this interface is created to represent APIs needed for archive-mode query (there will be a total 4-6 of them, 2 enabled in this PR, and more to enable in upcoming PR), so we don't need to add the `height` parameter to existing API, avoid changes like below
```
Account(address.Address, height uint64) (*iotextypes.AccountMeta, *iotextypes.BlockIdentifier, error)
```